### PR TITLE
[DEVOPS-703] Support storing builds in PRs

### PR DESCRIPTION
**Pull request by Chili-Man**
Removes the restriction on only allowing uploading builds when it is a merge request. Also deletes a lot of deprecated code.

### CODE REVIEW

File: s3deploy.sh [hunk 1]
```diff
@@ -249,45 +139,29 @@ s3d_exclude_paths() {
     export TARBALL_EXCLUDE_PATHS
 }
 
-# Uploads the tarball to s3.
+# Uploads the build tarball to s3, under the paths:
+# s3://og-deployments/$GIT_REPO_NAME/_global/$TRAVIS_COMMIT.tar.gz
```
Comment (mrlamroger) [inline s3deploy.sh]: should these say `_global_`

File: s3deploy.sh [hunk 2]
```diff
@@ -249,45 +139,29 @@ s3d_exclude_paths() {
...
-        # Upadate latest tarball
-        aws s3api copy-object --metadata-directive COPY --copy-source "$AWS_S3_BUCKET/$AWS_S3_OBJECT_PATH" --bucket "$AWS_S3_BUCKET" --key "$GIT_REPO_NAME/$TRAVIS_BRANCH/latest.tar.gz"
+    # Tar the build directory while excluding version control file
+    tar --exclude='./.git' $TARBALL_EXCLUDE_PATHS -c -z -f "$TARBALL_TARGET_PATH" .
 
-        # Copy to the global namespace and as its branch name. We will let the above still copy the builds into the old location while we migrate.
-        # Commit
-        aws s3api copy-object --metadata-directive COPY --copy-source "$AWS_S3_BUCKET/$AWS_S3_OBJECT_PATH" --bucket "$AWS_S3_BUCKET" --key "$AWS_S3_GLOBAL_OBJECT_PATH"
+    # Upload to S3
+    TARBALL_ETAG=$(ruby -e "require 'json'; resp = JSON.parse(%x[aws s3api put-object --acl private --bucket $AWS_S3_BUCKET --key $AWS_S3_GLOBAL_OBJECT_PATH --body $TARBALL_TARGET_PATH --metadata revision=$TRAVIS_COMMIT,pull_request=$TRAVIS_PULL_REQUEST,date=`date -u --iso-8601=seconds`]); puts resp['ETag'][1..-2]")
```
Comment (mrlamroger) [inline s3deploy.sh]: is there a way to heredoc this?

File: s3deploy.sh [hunk 3]
```diff
@@ -249,45 +139,29 @@ s3d_exclude_paths() {
...
 
-        # branch
-        aws s3api copy-object --metadata-directive COPY --copy-source "$AWS_S3_BUCKET/$AWS_S3_OBJECT_PATH" --bucket "$AWS_S3_BUCKET" --key "$GIT_REPO_NAME/$AWS_S3_GLOBAL_NAMESPACE_DIR/$TRAVIS_BRANCH.tar.gz"
-    else
-        echo 'Skipping build upload to S3'
+    # Copy to the global namespace as its branch name only if its not a pull request or not master, staging, and production branch.
```
Comment (mrlamroger) [inline s3deploy.sh]: Could you add a comment about what *should* be included. Sounds like PRs and all branches except master, staging, and production.

**Comment by jack749** [comment]
:+1:

**Pull request merged**